source/dotnet/Library/AdaptiveCards.Templating/AdaptiveCardsTemplateVisitor.cs

```diff
@@ -97,10 +97,15 @@ public AdaptiveCardsTemplateVisitor(Func<string, object> nullSubstitutionOption,
                     root = JToken.Parse(data);
                     PushDataContext(data, root);
                 }
-                catch (Exception e)
+                catch (JsonException e)
                 {
                     throw new Exception("Setting root data failed with given data context", e);
                 }
+                catch
+                {
+                    throw;
+                }
+                
```

Comment:
Probably just remove this catch block since we are only rethrowing here and nothing else.